File: Book Examples/AppLinking_completed/app/src/main/java/com/ebookfrenzy/applinking/AppLinkingActivity.java
package com.ebookfrenzy.applinking;

import android.content.Intent;
import android.net.Uri;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class AppLinkingActivity extends AppCompatActivity {

    public static final String LANDMARK_ID = "landmarkID";
    private static final String TAG = "AppIndexActivity";
    private EditText idText;
    private EditText titleText;
    private EditText descriptionText;
    private static final int PERSONAL = 1;
    private static final int PUBLIC = 0;
    MyDBHandler dbHandler = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_app_linking);

        idText = findViewById(R.id.idText);
        titleText = findViewById(R.id.titleText);
        descriptionText = findViewById(R.id.descriptionText);

        dbHandler = new MyDBHandler(this, null, null, 1);
    }

    public void findLandmark(View view) {

        if (!idText.getText().equals("")) {
            Landmark landmark = dbHandler.findLandmark(idText.getText().toString());

            if (landmark != null) {

                Intent intent = new Intent(this, LandmarkActivity.class);
                String landmarkid = idText.getText().toString();
                intent.putExtra(LANDMARK_ID, landmarkid);
                startActivity(intent);

                /*
                Log.i(TAG, "Found landmark = " + landmark.getTitle());
                Uri uri = Uri.parse("http://example.com/landmarks/" + landmark.getID());
                Intent intent = new Intent(Intent.ACTION_VIEW, uri);
                startActivity(intent);
                */
            } else {
                titleText.setText("No Match");
            }
        }
    }

    public void addLandmark(View view) {

        Landmark landmark =
                new Landmark(idText.getText().toString(), titleText.getText().toString(),
                        descriptionText.getText().toString(), 1);

        dbHandler.addLandmark(landmark);
        idText.setText("");
        titleText.setText("");
        descriptionText.setText("");

    }
}

File: KotlinGiraffeSamples/AppLinking/app/src/main/java/com/ebookfrenzy/applinking/AppLinkingActivity.kt
package com.ebookfrenzy.applinking

import android.net.Uri
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

import com.ebookfrenzy.applinking.databinding.ActivityAppLinkingBinding

class AppLinkingActivity : AppCompatActivity() {

    private lateinit var binding: ActivityAppLinkingBinding
    private var dbHandler: MyDBHandler? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAppLinkingBinding.inflate(layoutInflater)
        setContentView(binding.root)

        dbHandler = MyDBHandler(this, null, null, 1)
    }

    fun findLandmark(view: View) {

        if (binding.idText.text.toString() != "") {
            val landmark = dbHandler?.findLandmark(binding.idText.text.toString())

            if (landmark != null) {

                val intent = Intent(this, LandmarkActivity::class.java)
                val landmarkid = binding.idText.text.toString()
                intent.putExtra(LANDMARK_ID, landmarkid)
                startActivity(intent)

                /*
                val uri = Uri.parse("http://example.com/landmarks/" + landmark.id);
                val intent = Intent(Intent.ACTION_VIEW, uri);
                startActivity(intent);
                */
            } else {
                binding.titleText.setText("No Match")
            }
        }
    }

    fun addLandmark(view: View) {

        val landmark = Landmark(binding.idText.text.toString(), binding.titleText.text.toString(),
                binding.descriptionText.text.toString(), 1)

        dbHandler?.addLandmark(landmark)
        binding.idText.setText("")
        binding.titleText.setText("")
        binding.descriptionText.setText("")

    }

    companion object {

        val LANDMARK_ID = "landmarkID"
        private val TAG = "AppIndexActivity"
        private val PERSONAL = 1
        private val PUBLIC = 0
    }
}
